containerd-shim/process_linux.go

```diff
@@ -32,44 +33,52 @@ func (p *process) openIO() error {
 	p.stdinCloser = stdinCloser
 
 	if p.state.Terminal {
-		master, console, err := newConsole(uid, gid)
+		socket, err := runc.NewTempConsoleSocket()
 		if err != nil {
 			return err
 		}
-		p.console = master
-		p.consolePath = console
-		stdin, err := fifo.OpenFifo(ctx, p.state.Stdin, syscall.O_RDONLY, 0)
-		if err != nil {
-			return err
-		}
-		go io.Copy(master, stdin)
-		stdoutw, err := fifo.OpenFifo(ctx, p.state.Stdout, syscall.O_WRONLY, 0)
-		if err != nil {
-			return err
-		}
-		stdoutr, err := fifo.OpenFifo(ctx, p.state.Stdout, syscall.O_RDONLY, 0)
-		if err != nil {
-			return err
-		}
-		p.Add(1)
-		p.ioCleanupFn = func() {
-			master.Close()
-			stdoutr.Close()
-			stdoutw.Close()
-		}
-		go func() {
-			io.Copy(stdoutw, master)
-			p.Done()
+		p.socket = socket
+
+		go func() error {
+			master, err := socket.ReceiveMaster()
+			if err != nil {
+				return err
+			}
+			p.console = master
+			stdin, err := fifo.OpenFifo(ctx, p.state.Stdin, syscall.O_RDONLY, 0)
+			if err != nil {
+				return err
+			}
+			go io.Copy(master, stdin)
+			stdoutw, err := fifo.OpenFifo(ctx, p.state.Stdout, syscall.O_WRONLY, 0)
+			if err != nil {
+				return err
+			}
+			stdoutr, err := fifo.OpenFifo(ctx, p.state.Stdout, syscall.O_RDONLY, 0)
+			if err != nil {
+				return err
+			}
+			p.Add(1)
+			p.ioCleanupFn = func() {
+				master.Close()
+				stdoutr.Close()
+				stdoutw.Close()
+			}
+			go func() {
+				io.Copy(stdoutw, master)
+				p.Done()
+			}()
+			return nil
 		}()
```

Comment:
Ya, i think you are right, i'm going to have to add some sync around here, i'm seeing some issues in docker but having been able to replicate with containerd only right now